assets/prototype/application/hooks/useIfMounted.ts

```diff
@@ -0,0 +1,19 @@
+import { useRef, useEffect, useCallback } from '@wordpress/element';
+
+const useIfMounted = () => {
+	const isMounted = useRef(true);
+	useEffect(
+		() => () => {
+			isMounted.current = false;
+		},
+		[]
+	);
+
+	const ifMounted = useCallback((func) => {
+		if (isMounted.current && func) {
```

Comment:
need to either convert this to typescript and cast `func` as a Function, or add a typeof check here